oracle_fdw.c

```diff
@@ -821,9 +822,63 @@ oracleGetForeignRelSize(PlannerInfo *root, RelOptInfo *baserel, Oid foreigntable
 				fdwState->oraTable->cols[i]->used = 1;
 	}
 
-	/* construct Oracle query and get the list of parameters and actions for RestrictInfos */
-	fdwState->query = createQuery(fdwState, baserel, for_update, root->query_pathkeys);
-	elog(DEBUG1, "oracle_fdw: remote query is: %s", fdwState->query);
+	/* createQuery is moved to GetForeignPlan. */
+	/* fdwState->for_update is needed by createQuery in GetForeignPlans */
+	fdwState->for_update = for_update;
+	/* fdwState->usable_pathkeys is needed by create_foreign_scan in GetForeignPaths. */
+	/* So it is created in this place. */
+
+	/*
+	 * Determine whether we can potentially push query pathkeys to the remote
+	 * side, avoiding a local sort.
+	 */
+	ListCell *cell;
```

Comment:
This causes a compilation warning:

    oracle_fdw.c: In function ‘oracleGetForeignRelSize’:
    oracle_fdw.c:835: warning: ISO C90 forbids mixed declarations and code